packages/nimble-components/src/number-field/tests/number-field.spec.ts

```diff
@@ -75,6 +79,123 @@ describe('NumberField', () => {
         processUpdates();
         expect(element.control.getAttribute('aria-required')).toBe('false');
     });
+
+    it('disallows input of characters that cannot be used in a number', () => {
+        page.setText('0123+-.abcdeABCDE,!@#$%^&*()_[]{}|;:\'"<>?/456789');
+        expect(page.getDisplayValue()).toEqual('0123+-.e456789');
+    });
+
+    const badInputCases = [
+        { name: '01...2', expectedValue: '1' },
+        { name: '0...2', expectedValue: '0' },
+        { name: '...2', expectedValue: '' },
+        { name: '.2', expectedValue: '0.2' },
+        { name: '++1', expectedValue: '' },
+        { name: '+1', expectedValue: '1' },
+        { name: '--1', expectedValue: '' },
+        { name: '-1', expectedValue: '-1' },
+        { name: '1.12ee4', expectedValue: '1.12' },
+        { name: '1.12e4', expectedValue: '11200' },
+        { name: '112e4', expectedValue: '1120000' }
```

Comment:
Why are these "bad input" cases? Seems like  exponential notation works? Same with ".2", "-1"

Maybe these are just "input cases" instead of bad cases?